psi4/src/psi4/libfock/jk.cc

```diff
@@ -71,7 +71,12 @@ JK::~JK() {}
 std::shared_ptr<JK> JK::build_JK(std::shared_ptr<BasisSet> primary, std::shared_ptr<BasisSet> auxiliary,
                                  Options& options, std::string jk_type) {
 
-    if (options.get_str("SCREENING") == "DENSITY" && !(jk_type == "DIRECT" || options.get_bool("DF_SCF_GUESS"))) {
+    bool do_density_screen = options.get_str("SCREENING") == "DENSITY";
+    bool do_df_scf_guess = options.get_bool("DF_SCF_GUESS");
+    
+    bool can_do_density_screen = (jk_type == "DIRECT" || jk_type == "LINK");
+
+    if (do_density_screen && !(can_do_density_screen || do_df_scf_guess)) {
```

Comment:
Why is it important to check `DF_SCF_GUESS`? It's unclear to me which keywords apply to which stages of the computation.

Reply:
This part is basically a cleanup of a code part that was previously present in Psi4, but rather messy. The basic idea is to exit out if trying to use density screening in conjunction with an algorithm that doesn't actually support it. The DF_SCF_GUESS option opens up the possibility of combining an integral-direct algorithm that supports density screening, with DFJK, which doesn't. Thus, a check for DF_SCF_GUESS would be required here.